api/proquest/importer.py

```diff
@@ -0,0 +1,647 @@
+                book = self._get_book(
+                    patron, configuration, licensepool.identifier.identifier
+                )
+
+                if book.content is not None:
+                    fulfillment_info = FulfillmentInfo(
+                        licensepool.collection,
+                        licensepool.data_source.name,
+                        licensepool.identifier.type,
+                        licensepool.identifier.identifier,
+                        content_link=None,
+                        content_type=internal_format.delivery_mechanism.media_type,
+                        content=book.content,
+                        content_expires=None,
+                    )
+                else:
+                    fulfillment_info = FulfillmentInfo(
+                        licensepool.collection,
+                        licensepool.data_source.name,
+                        licensepool.identifier.type,
+                        licensepool.identifier.identifier,
+                        content_link=book.link,
+                        content_type=internal_format.delivery_mechanism.media_type,
+                        content=None,
+                        content_expires=None,
+                    )
+
+                self._logger.info(
+                    "Finished fulfilling {0} for patron {1}: {2}".format(
+                        internal_format, patron, fulfillment_info
+                    )
+                )
+
+                return fulfillment_info
+        except BaseError as exception:
+            raise CannotFulfill(six.ensure_text(str(exception)))
+
+    def external_integration(self, db):
+        """Return an external integration associated with this object.
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :return: External integration associated with this object
+        :rtype: core.model.configuration.ExternalIntegration
+        """
+        return self.collection.external_integration
+
+
+class ProQuestOPDS2ImportMonitor(OPDS2ImportMonitor, HasExternalIntegration):
+    PROTOCOL = ExternalIntegration.PROQUEST
+
+    def __init__(
+        self,
+        client_factory,
+        db,
+        collection,
+        import_class,
+        force_reimport=False,
+        **import_class_kwargs
+    ):
+        """Initialize a new instance of ProQuestOPDS2ImportMonitor class.
+
+        :param client_factory: ProQuest API client
+        :type client_factory: api.proquest.client.ProQuestAPIClientFactory
+        """
+        super(ProQuestOPDS2ImportMonitor, self).__init__(
+            db, collection, import_class, force_reimport, **import_class_kwargs
+        )
+
+        self._client_factory = client_factory
+        self._feeds = None
+        self._client = self._client_factory.create(self)
+
+        self._logger = logging.getLogger(__name__)
+
+    def _parse_feed(self, feed, silent=True):
```

Comment:
It's a snippet from `OPDS2Importer` class. I'll refactor it when I'll be working on the code allowing to set up a custom classifier

Reply:
Can you go into more detail about a custom classifier? I don't think import is the appropriate place to set up a custom classifier, because import only happens once and classification happens many times over the lifespan of a book, as the classification rules improve.

Reply:
Sorry, I missed this comment. Sure, you're absolutely right. I meant that I wanted to do this refactoring (removing duplicated code) together with changing the default classifier used for the ProQuest feed to LCSH because both of them require changes in `server_core` and I would wan't to wait that long